px/bird/templates/_helpers.tpl

```diff
@@ -47,6 +47,26 @@ alert-tier: px
 alert-service: px
 {{- end }}
 
+# this definition will spread pods for any domain and service accross 
+# all available AZs - ignores looking glass pods
+{{- define "bird.topology_spread" }}
+- maxSkew: 1
```

Comment:
This tries to evenly balance out the pods over all zones, right? I think the problem here is the broad label selector

Consider the following placement:
```
AZ A
routeserver-v4-service-1-domain-1-1
routeserver-v4-service-1-domain-2-1
routeserver-v4-service-2-domain-1-1
routeserver-v4-service-2-domain-2-1
routeserver-v6-service-1-domain-1-1
routeserver-v6-service-1-domain-2-1
routeserver-v6-service-3-domain-1-2

AZ B
routeserver-v4-service-1-domain-1-2
routeserver-v4-service-1-domain-2-2
routeserver-v4-service-3-domain-1-1
routeserver-v4-service-3-domain-2-1
routeserver-v6-service-1-domain-1-2
routeserver-v6-service-1-domain-2-2
routeserver-v6-service-3-domain-2-2


AZ C
routeserver-v4-service-2-domain-1-2
routeserver-v4-service-2-domain-2-2
routeserver-v4-service-3-domain-1-2
routeserver-v4-service-3-domain-2-2
routeserver-v6-service-3-domain-1-1
routeserver-v6-service-3-domain-2-1


```
Good old `AntiAffinity` is respected, no pod of the same service and same domain runs in the same zone.
`topologySpread` is also respected, skew is `1` (if I understood the docs correctly).

However, only service 3 in the v6 address family is spread over 3 AZs. Service 1 in both AFIs runs only in A and B, service 2 in both AFIs runs in A and C, service 3 in the v4 AFI runs in B and C. Since each PX service and AFI provides an independent function, we didn't really gain any redundancy here (of course this is the most pessimistic scenario).

That means,  `matchExpressions` needs another mach on `px.cloud.sap/service` similar like we do in the template `bird.domain.affinity`. We would also need a match on the address family.